src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,298 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager* AnalyzerManager::m_pAnalyzerManager = NULL;
+
+//static
+AnalyzerManager& AnalyzerManager::getInstance(UserSettingsPointer pConfig) {
+    if (!m_pAnalyzerManager) {
+        // There exists only one UserSettingsPointer in the app, so it doens't matter
+        // if we only assign it once.
```

Comment:
The implementation needs that there exists only one manager so that it can control the number of threads in execution. Since it is used from two different places (analysisfeature and playermanager), it cannot be an instance on those classes either.

Still, I think that I was wrong in using the getInstance() naming/model and should have been a MainApplication instance, where I have UserSettingsPointer.

I was busy enough trying to understand QThread and signals so I took the easiest path. Now I am open to change this if we give me the correct guidelines to do so.